engine-precompiles/src/native.rs

```diff
@@ -395,30 +440,44 @@ impl<I: IO> Precompile for ExitToNear<I> {
             erc20_address,
             amount: types::u256_to_arr(&exit_event.amount),
         };
-        #[cfg(feature = "error_refund")]
-        let refund_promise = PromiseCreateArgs {
-            target_account_id: refund_on_error_target,
-            method: "refund_on_error".to_string(),
-            args: refund_args.try_to_vec().unwrap(),
-            attached_balance: Yocto::new(0),
-            attached_gas: costs::REFUND_ON_ERROR_GAS,
+
+        let precompile_call_args = ExitToNearPrecompileCallbackCallArgs {
+            #[cfg(feature = "error_refund")]
+            refund: Some(refund_args),
+            #[cfg(not(feature = "error_refund"))]
+            refund: None,
+            transfer_near: transfer_near_args,
         };
+
+        let callback_promise =
+            if precompile_call_args == ExitToNearPrecompileCallbackCallArgs::default() {
```

Comment:
nitpick: Seems a bit cleaner the following way, but your call:
```rust
let is_exit_to_near_precompile = precompile_call_args != ExitToNearPrecompileCallbackCallArgs::default();
let callback_promise = is_exit_to_near_precompile.then(|| PromiseCreateArgs {
    target_account_id: self.current_account_id.clone(),
    method: "exit_to_near_precompile_callback".to_string(),
    args: precompile_call_args.try_to_vec().unwrap(),
    attached_balance: Yocto::new(0),
    attached_gas: costs::EXIT_TO_NEAR_CALLBACK_GAS,
});
```

Reply:
I've moved this part of code down to improve the readability 870f97a9d33b648be8f7e1b0d0978eea05d0fb69